clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -328,6 +328,8 @@ static void collectSYCLAttributes(Sema &S, FunctionDecl *FD,
                SYCLIntelNoGlobalWorkOffsetAttr, SYCLSimdAttr>(A);
   });
 
+  // TODO: ERICH: We should probably warn on the bottom3 as well and turn this
```

Comment:
This is one I'll probably visit once I get the build-bots to agree I'm OK with my refactor so far.  It seems that we should still be warning instead of silently dropping here, but even if that is a problem, I still want to make it so we have much less copy/paste here.

Reply:
Ugh, we actually have a number of lit tests that depend on us not diagnosing or removing the bottom 3 (now 4 after merging @zahiraam s patch).  I simplified this now, but we can't combine all 4.  That is perhaps a future "TODO".